### PR TITLE
Advanced import completion

### PR DESCRIPTION
**Pull request by etiennerichart**
when typed: from <module> import <> and tab is pressed will background import the module to get better autocomplete options. Note: If nothing is yet typed after the import the autocomplete will not show any options.

### CODE REVIEW

**Comment by sebastinas** [comment]
I'm not sure what this PR tries to achieve. What's the behavior before and after this change?

**Comment by thomasballinger** [comment]
The goal of this PR is that when typing out an import for a module (numpy for example)
`>>> from numpy import a<tab>` should display all submodules and module attributes. Hitting tab might cause numpy to be imported too.

I don't know the status of this PR, it's been on my list to review for a bit; I worked with @etiennerichart on it but haven't been paying attention for a couple months now.

**Comment by sebastinas** [comment]
Ah, yes, that's indeed useful.

File: bpython/curtsiesfrontend/repl.py [hunk 1]
```diff
@@ -897,6 +906,20 @@ def only_whitespace_left_of_cursor():
                 self.add_normal_character(" ")
             return
 
+        # 5. if line is `from a import` and tab then import module and attr
+        line = self.current_line.split()
+        module_name = ""
+        if "from" in line and "import" in line:
+            # returns <module> name if the line contains: "from <module> import"
+            module_name = from_import_tab(self.current_line)
+            print(module_name)
```
Comment (sebastinas) [inline bpython/curtsiesfrontend/repl.py:915]: This is a left over debug print, I guess.

**Comment by sebastinas** [comment]
We just removed 0.20 to fix compatibility with Python 3.9 and now dropped support for Python 2 in master. The latter touched almost the whole code base. Could you please rebase this PR on top of current master? Thanks

File: bpython/line.py [hunk 2]
```diff
@@ -120,6 +120,19 @@ def current_object_attribute(cursor_offset, line):
     return None
 
 
+from_import_tab_re = LazyReCompile(r"from ([\w0-9_.]*)\s+import")
+
+
+def from_import_tab(line):
+    matches = from_import_tab_re.finditer(line)
+    try:
+        module = matches.__next__()
```
Comment (sebastinas) [inline bpython/line.py:129]: This should be `next(matches)`, right?

File: bpython/importcompletion.py [hunk 3]
```diff
@@ -99,6 +100,19 @@ def module_attr_matches(name):
     return attr_matches(name, prefix="", only_modules=True)
 
 
+def try_to_import(module_name):
```
Comment (sebastinas) [inline bpython/importcompletion.py:103]: Is this really necessary? Shouldn't those modules already be found by our normal module processing routines?

**Comment by etiennerichart** [comment]
I have not been able to test functionality yet since I'm having trouble starting bpython on my PC. I'm getting the following error:
`File "c:\users\etien\appdata\local\programs\python\python38-32\lib\site-packages\blessings\__init__.py", line 5, in <module>
    import curses
  File "c:\users\etien\appdata\local\programs\python\python38-32\lib\curses\__init__.py", line 13, in <module>
    from _curses import *
ModuleNotFoundError: No module named '_curses'`
If someone wants to test it for me, just try:
`from <module> import <at least one letter>` seeing if there are autocomplete suggestions, then pressing tab to see if you get more suggestions. I usually test on using numpy module but you may have to pip install it.

**Comment by sebastinas** [comment]
As far as I can tell, the PR works as expected. I'm just not sure if we need the additional module discovery code.

**Comment by sebastinas** [comment]
Okay, I took a look at this PR again. I don't think we should merge it in its current state. `from x import y` auto completion is already in handled in `bpython.importcompletion` and `bpython.autocomplete`. The issue that this PR tries to solve is auto completion when our import completion hasn't processed `x`. What we could discuss if we can fix the import completion code to process `x` in `bpython.importcompletion.complete` if it is not yet known.

But I don't think we should add a curtsies specific workaround that breaks abstraction.

**Pull request closed without merging**